src/commands/init.ts

```diff
@@ -1,58 +1,71 @@
-import { GluegunCommand, GluegunToolbox } from 'gluegun';
+import { GluegunCommand, GluegunPrint, GluegunToolbox } from 'gluegun';
 import * as fsExtra from 'fs-extra';
 import * as path from 'path';
 
 const ignorePath = [
-  'unthink-stack/lib/',
-  'unthink-stack/public/',
-  'unthink-stack/node_modules/',
+  // Exclude specific files
+  /.*unthink-stack\/\.env$/,
+
+  // Exclude lib, public, node_modules
+  /.*unthink-stack\/lib($|\/.*)/,
+  /.*unthink-stack\/public($|\/.*)/,
+  /.*unthink-stack\/node_modules($|\/.*)/,
 ];
 
+const projectNamePattern = /^[a-z0-9-]+$/;
+
+function printProjectNameError(print: GluegunPrint, additionalMessage: string): void {
+  const projectNameRequirements = [
+    additionalMessage,
+    '',
+    'Project name must be lower case containing only letters (a-z), numbers (0-9) and dashes (-).',
+    'The name can contain the full path to the destination folder which will be the project name rules still apply.',
+    '',
+    'example: foo/bar/my-new-project.'
+  ];
+
+  for (const line of projectNameRequirements) {
+    print.error(line);
+  }
+}
 
 const command: GluegunCommand = {
-  name: 'init',
-  alias: ['i'],
+  name: 'initialize',
+  alias: ['i', 'init'],
   description: 'Start a new project',
 
   run: async (toolbox: GluegunToolbox): Promise<void> => {
-    const projectName = toolbox.parameters.first;
-    const targetPath = `./${projectName}`;
-    const baseStackDir = path.join(__dirname, '../../unthink-stack');
+    const projectPath = toolbox.parameters.first;
 
-    try {
-      const existingFiles = await fsExtra.readdir(targetPath);
-      if (existingFiles.length > 0) {
-        toolbox.print.error(`${ path.resolve(targetPath) } is not empty.`);
-        return;
-      }
-    } catch (e) {
-      // lets just assume it's not a directory and move on.
+    if (!projectPath) {
+      printProjectNameError(toolbox.print, 'Project name not specified.');
+      return;
     }
 
-    try {
-      await fsExtra.ensureDir(targetPath);
-    } catch (e) {
-      toolbox.print.error(`${ path.resolve(targetPath) } is not empty.`);
+    const projectName = path.basename(projectPath);
+    if (!projectNamePattern.test(projectName)) {
+      printProjectNameError(toolbox.print, `${projectName} is invalid.`);
       return;
     }
 
+    const targetPath = `${projectPath}`;
+    const baseStackDir = path.join(__dirname, '../../unthink-stack');
+
+    // If path exists bail even if it's just an empty directory
```

Comment:
I would like to think about allowing a "force" to create a project in an existing directory. Reason being, the easiest way to get a project into a repo  is by creating the repo, cloning, and then creating the project in it.

Reply:
Yeah I think that is a good way to handle that. By default it's "safe" and the "force" option lets you say "I know what I'm doing".

Reply:
Simple enough. I also added a prompt to let the user back out when using `--force`

usage:

```bash
unthink init some/existing/dir --force
? Are you sure you want to force initialization? (overwrites files in destination) (y/Initialized project
✔ Are you sure you want to force initialization? (overwrites files in destination) (y/N) · true

```

Prompt is a bit weird as noted above with the arrow. I don't see options to change that but specifically don't know why it's adding initialized project

```bash
... in destination) (y/Initialized  <-- here
```